packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -502,9 +515,16 @@ export class StripeHelper {
    * - remove the cache entry
    */
   async removeCustomer(uid: string, email: string) {
-    const customer = await this.fetchCustomer(uid, email);
-    if (customer) {
-      await this.stripe.customers.del(customer.id);
+    const customer = await getAccountCustomerByUid(uid);
+    if (customer && customer.stripeCustomerId) {
+      await this.stripe.customers.del(customer.stripeCustomerId);
+      const recordsDeleted = await deleteAccountCustomer(uid);
```

Comment:
If this fails, will we have enough information to locate the user manually to delete the record?

Reply:
Yes. If you look at the check below: if there are no records affected (the delete calls returns an integer indicating the number of rows that were removed), the user's account id is logged along with a message indicating that the item wasn't removed.